app/Repositories/SongRepository.php

```diff
@@ -230,4 +231,13 @@ public function getRandomByGenre(string $genre, int $limit, ?User $scopedUser =
             ->inRandomOrder()
             ->get();
     }
+
+    /**
+     * @param User $user
+     * @return SongBuilder
+     */
+    public function getSongsByPlaylist(User $user): SongBuilder
```

Comment:
The method name is a lie. This method doesn't return songs by a playlist, but a `SongBuilder`. Also, out of scope.